### ISSUE TITLE
Cookies API

### ISSUE THREAD

**Issue by kamesh-a** [atom/electron#1606]
Hello Everyone,

I am a newbie to Electron (Atom-shell), Was figuring out the api provided in ATOM shell. While doing mock up codes. I was unable to find any api to retrieve cookies and delete specific cookies for domain.

Am i missing something, can someone put me in a right direction.
```
Mac : ~/Library/Application Support/<appName>/Cookies i deleted all were gone.
windows : appData/<appName>/cookies 

```

deletion will delete everything But is there any api to collect all Cookies and manage selective domains to delete them? or can we enable https://developer.chrome.com/extensions/cookies api

BTW, I just love the architecture and way atom is working embedding CEF and Node

Cheers!!!!!

**Comment by kamesh-a** [comment]
do we have any estimated plan on embedding this?

**Comment by yoshuawuyts** [comment]
Having cookies would be nice. The current workaround is to use [`electron-cookies`](https://github.com/hstove/electron-cookies), which implements cookies on top of `localStorage`.

**Comment by yoshuawuyts** [comment]
Also possibly duplicate of https://github.com/atom/electron/issues/1142 (closed)

**Comment by kamesh-a** [comment]
Thanks, Will do some mock test codes..

**Comment by kamesh-a** [comment]
Is it possible, how can we do a bulk flush of data for example, getAll, removeAll, if don't know the cookie name used or loading third party websites?

**Comment by hokein** [comment]
@kamesh-a
You can pass an empty `details` to `WebContents.session.cookies.get` for `getAll` function. And once you get each cookies, you can delete it iteratively by using `WebContents.session.cookies.remove` API.

[electron-sample-apps](https://github.com/hokein/electron-sample-apps/) has provided a cookies demo.

**Comment by kamesh-a** [comment]
Great thanks @hokein sample snippet worked.

var win = require('remote').getCurrentWindow();

win.webContents.session.cookies.get({}, function(error, cookies) {
    if (error) throw error;
    for (var i = cookies.length - 1; i >= 0; i--) {
        removeCookie(cookies[i])
    };
});


function removeCookie(cookie) {
    var url = "http" + (cookie.secure ? "s" : "") + "://" + cookie.domain +cookie.path;
    win.webContents.session.cookies.remove({
            "url": url,
            "name": cookie.name
        },
        function(error) {
            if (error) throw error;
            console.log('cookie delete : ', cookie);
        });
}

**Comment by anupshinde** [comment]
Is it possible or allowed to persist http-only session cookies using the "persist:" flag (in webview)?

I tried the same with github and gmail. That works. Next time I load the electron app, it loads my accounts.

But I am embedding a custom web application within a web-view that uses http-only session cookies. And even with the persist flag, the session is not loaded.  Something like this in the response headers.

Set-Cookie:session=xxxxxx-xxxx-xxxx-xxxx-xxxxxxxx; HttpOnly; Path=/

When I use Chrome with the custom web app, it reuses existing session - probably storing the session cookies in some way.

I did a workaround to store the cookies post login and the pass the session cookies using "...cookies.set(...)". That works, but just wondering if there is a way do avoid this kind of coding.

(I am new to electron)
